src/core/tsi/alts_transport_security.cc

```diff
@@ -63,3 +64,15 @@ void grpc_tsi_alts_shutdown() {
   gpr_cv_destroy(&g_alts_resource.cv);
   gpr_mu_destroy(&g_alts_resource.mu);
 }
+
+void grpc_tsi_alts_shared_resource_ref() { gpr_ref(&g_alts_resource.refcount); }
+
+void grpc_tsi_alts_shared_resource_unref() {
+  if (gpr_unref(&g_alts_resource.refcount)) {
+    grpc_completion_queue_shutdown(g_alts_resource.cq);
```

Comment:
Lines 72-76 and Lines 58-62 can be refactored into a single function to avoid code duplication